### PR TITLE
Rchain 3548 slash invalid validator approve evil block

### PR DESCRIPTION
**Pull request by zsluedem**
## Overview
<sup>_What this PR does, and why it's needed_</sup>

### JIRA ticket:
https://rchain.atlassian.net/browse/RCHAIN-3548

### Notes
<sup>_Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else._</sup>

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

**Comment by zsluedem** [comment]
bors try

**Comment by zsluedem** [comment]
bors try

File: integration-tests/test/test_slash.py [hunk 1]
```diff
@@ -155,11 +157,94 @@ def test_slash_justification_not_correct(command_line_options: CommandLineOption
...
+
+        client.send_block(block_not_slash_invalid_block, validator3)
+
+        # Because validator2 doesn't slash validator1's block while validator3 slash validator1's block,
+        # hence there are some comm events lack of in validator3 which cause an invalid transaction
+        record_invalid = re.compile("Recording invalid block {}... for InvalidTransaction".format(invalid_block_hash.hex()[:10]))
+        wait_for_log_match(context, validator3, record_invalid)
+
+
+        validator3.deploy(contract, BOOTSTRAP_NODE_KEY)
```
Comment (woky) [inline integration-tests/test/test_slash.py:251]: Do you use `BOOTSTRAP_NODE_KEY` here just as a _random_ deployer key?

**Review by woky — APPROVED** [review]
So this does the following?
1. v1 proposes valid block
2. v1 creates another block with invalid parent block hash and sends it to v2
3. v2 creates block using v1's second block as parent hash (i.e. no slashing, another invalid block) and sends it to v3
4. v3 records invalid block (`InvalidTransaction`) from v2
5. v3 proposes block which slashes both v1 and v2

File: integration-tests/test/test_slash.py (hunk 1 above)
Comment (zsluedem) [reply integration-tests/test/test_slash.py:251]: > So this does the following?
> 
> 1. v1 proposes valid block
> 2. v1 creates another block with invalid parent block hash and sends it to v2
> 3. v2 creates block using v1's second block as parent hash (i.e. no slashing, another invalid block) and sends it to v3
> 4. v3 records invalid block (`InvalidTransaction`) from v2
> 5. v3 proposes block which slashes both v1 and v2

You are right!



> Do you use `BOOTSTRAP_NODE_KEY` here just as a _random_ deployer key?

Actually it doesn't matter which key as long as validator3 proposed a new block.

Comment (adaszko) [reply integration-tests/test/test_slash.py:251]:
```v1 proposes valid block
v1 creates another block with invalid parent block hash and sends it to v2
v2 creates block using v1's second block as parent hash (i.e. no slashing, another invalid block) and sends it to v3
v3 records invalid block (InvalidTransaction) from v2
v3 proposes block which slashes both v1 and v2
```

Can we have this put into the docstring?  This is a great summary of what the test scenario does.

**Review by adaszko — APPROVED** [review]

File: integration-tests/test/test_slash.py (hunk 1 above)
Comment (adaszko) [reply integration-tests/test/test_slash.py:251]: Thanks 👍

**Pull request merged**